### PR TITLE
Simplify nn.log_softmax and add edit_excluded argument.

### PR DESCRIPTION
**Pull request by carlosgmartin**
This commit simplifies [log_softmax](https://docs.jax.dev/en/latest/_autosummary/jax.nn.log_softmax.html) and adds an `edit_excluded` argument to control whether to set excluded elements to `-jnp.inf` (which is not always desirable).

### CODE REVIEW

**Comment by jakevdp** [comment]
I don't entirely understand the purpose of this new parameter. Do you have a simple example of where it might be useful?

**Comment by carlosgmartin** [comment]
These `-inf` values seem to sometimes cause `nan`s downstream under `grad`. (See https://github.com/jax-ml/jax/issues/1052 and linked issues there.) Thus it would help to just ignore the masked out logits from calculations (using the same `where` mask at each stage of the processing pipeline) rather than setting the corresponding elements to `-inf`.

**Comment by jakevdp** [comment]
I'm not convinced that use-case warrants the introduction of a new parameter to `logsoftmax`.

Perhaps you could use `jnp.where` after the fact to set the `-inf` values to something different?
